src/main/java/lesson2/problem7/ReverseList3Pointer.java

```diff
@@ -0,0 +1,23 @@
+package lesson2.problem7;
+
+import lesson2.problem0.Link;
+
+public class ReverseList3Pointer {
+
+    public Link reverseList3Pointer(Link head) {
+        if (head == null || head.next == null){
+            return head;
+        }
+        Link pre = null;
+        Link cursor = head;
+        Link next = cursor.next;
+        while(next != null){
+            cursor.next = pre;
+            pre = cursor;
+            cursor = next;
+            next = next.next;
```

Comment:
In that case there will be if condition to check if the next is null here, so you will not get NPE.